src/pages/cli/graphql/authorization-rules.mdx

```diff
@@ -0,0 +1,368 @@
+
+**Known limitation**: Real-time subscriptions are not supported for dynamic group authorization.
+
+### Custom authorization rule
+
+You can define your own custom authorization rule with a Lambda function.
+
+```graphql
+type Salary @model @auth(rules: [{ allow: custom }]) {
+  id: ID!
+  wage: Int
+  currency: String
+}
+```
+
+The Lambda function of choice will receive an authorization token from the client and execute the desired authorization logic. The AppSync GraphQL API will receive a payload from Lambda after invocation to allow or deny the API call accordingly.
+
+Configure the GraphQL API with the Lambda authorization mode:
+
+```bash
+amplify update api
+```
+```
+? Select a setting to edit: 
+> Authorization modes
+
+> Lambda
+
+? Choose a Lambda source:
+> Create a new Lambda function
+```
+
+You can use the default Amplify provided template as a starting point for your custom authorization rule. The authorization Lambda function receives:
+
+```json
+{
+    "authorizationToken": "ExampleAuthToken123123123", # Authorization token specified by client
+    "requestContext": {
+        "apiId": "aaaaaa123123123example123", # AppSync API ID
+        "accountId": "111122223333", # AWS Account ID
+        "requestId": "f4081827-1111-4444-5555-5cf4695f339f",
+        "queryString": "mutation CreateEvent {...}\n\nquery MyQuery {...}\n", # GraphQL query
+        "operationName": "MyQuery", # GraphQL operation name
+        "variables": {} # any additional variables supplied to the operation
+    }
+}
+```
+
+Your Lambda authorization function needs to return the following JSON:
+
+```json
+{
+  // required
+  "isAuthorized": true, // if "false" then an UnauthorizedException is raised, access is denied
+  
+  // optional
+  "deniedFields": ["TypeName.FieldName"], // Forces the fields to "null" when returned to the client
+  "resolverContext": { "banana":"very yellow" }, // JSON object visible as $ctx.identity.resolverContext in VTL resolver templates
+  "ttlOverride": 10 // The number of seconds that the response should be cached for. Overrides default specified in "amplify update api"
+}
+```
+
+Review the Amplify Library documentation to set the custom authorization token for [GraphQL API](/lib/graphqlapi/authz/q/platform/js/#aws-lambda) and [DataStore](/lib/datastore/setup-auth-rules/q/platform/js/#configure-custom-authorization-logic-with-aws-lambda).
+
+## Configure multiple authorization rules
+When combining multiple authorization rules, they are "logically OR"-ed. 
+
+```graphql
+type Post @model @auth(rules: [
+  { allow: public, operations: [read], provider: iam },
+  { allow: owner }]) {
+  title: String
+  content: String
+}
+```
+
+In the example above:
+- any user (signed in or not, verified by IAM) is allowed to read all posts
+- owners are allowed to create, read, update, and delete their own posts.
+
+## Field-level authorization rules
+
+TODO ADD SUBSCRIPTION BEHAVIOR
```

Comment:
v2 does allow required (aka non-null) fields though it will require that allow entities (mentioned in the auth rules) have read access to the required field. Ref: [Error Message](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts#L355-L358)

Reply:
Addressed in latest update